opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -144,9 +143,7 @@ def __init__(
         attributes: types.Attributes = None,
     ) -> None:
         super().__init__(context)
-        self._attributes = BoundedAttributes(
-            attributes=attributes
-        )  # type: types.Attributes
+        self._attributes = attributes
```

Comment:
Why did BoundedAttributes get removed?

Reply:
`attributes` are already passed as BoundedAttributes. Also, this makes the `Link` class implementation consistent with `Event` class for attributes.

Reply:
It should be `BoundedAttributes` class in `Event` class also. Don't passdown as `BoundedAttributes` but revert it back creating one in the Link class. The bounded condition should apply even when `Link` directly is used.

Reply:
If I just passdown the attributes received and not `BoundedAttributes`, then how will I pass the `max_len` and `max_len_value` constraints? Should I accept these as well in `Link.__init__()`?

Reply:
I see the existing design makes it non-trivial to enforce limits in each class on their own. The main goal is to make `attributes` input a `BoundedAttributes` so that we can enforce limits such as `OTEL_ATTRIBUTE_COUNT_LIMIT` etc.. All of the details should be internal and shouldn't be part of the constructor.  I don't want to scope creep this PR beyond add_link. So this is fine.